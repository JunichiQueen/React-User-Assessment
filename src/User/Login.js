import React, { Component } from 'react';
import axios from 'axios';

export default class Login extends Component{
    constructor(){
        super();
        this.state={
            success: "",
            error: ""
        }
    }
    login = (e) => {
        e.preventDefault();
        let name = e.target[0].value;
        let password = e.target[1].value;

        axios
        .get(`http://localhost:5000/user/name/${name}/${password}`).then(response => {
            if (response.data.Status.toString() == "Logged In"){
                this.setState({
                    success: response.data.Status.toString(),
                })
            }
            if (response.data.Status.toString() == "Not Logged In") {
                this.setState({
                    error: response.data.Status.toString()
                })
            }
        }).catch(err => {
            this.setState({
                error: "Bad Username"
            })
        });
    }
    render(){
        return(
            <div>
                <h2 className="text-center jumbotron">Login</h2>
                <form className="text-center jumbotron" onSubmit={this.login}>
                    <p>Username</p>
                    <input></input>
                    <p>Password</p>
                    <input></input>
                    <br></br>
                    <button type="submit">Login</button>
                    <p style={{color: 'green'}}>{this.state.success}</p>
                    <p style={{color: 'red'}}>{this.state.error}</p>
                </form>
            </div>
        )
    }
}
