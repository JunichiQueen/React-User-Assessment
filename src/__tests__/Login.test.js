import React from 'react';
import ReactDOM from 'react-dom';
import Login from '../User/Login';
import { shallow } from "enzyme";
import renderer from 'react-test-renderer'
import "../setupTests"



it('renders without crashing', () => {
  shallow(<Login />);
});

it('snapshot app', () => {
  const wrapper = renderer.create(<Login />).toJSON();
  expect(wrapper).toMatchSnapshot();
});