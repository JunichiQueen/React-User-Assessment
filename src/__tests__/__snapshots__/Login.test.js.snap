// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot app 1`] = `
<div>
  <h2
    className="text-center jumbotron"
  >
    Login
  </h2>
  <form
    className="text-center jumbotron"
    onSubmit={[Function]}
  >
    <p>
      Username
    </p>
    <input />
    <p>
      Password
    </p>
    <input />
    <br />
    <button
      type="submit"
    >
      Login
    </button>
    <p
      style={
        Object {
          "color": "green",
        }
      }
    >
      
    </p>
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </p>
  </form>
</div>
`;
